pkg/cluster/wait.go

```diff
@@ -51,6 +52,10 @@ func WaitFor(ctx context.Context, client kubernetes.Reader, cluster *anywherev1.
 
 		observedGeneration := c.Status.ObservedGeneration
 		generation := c.Generation
+
+		log := logger.Get()
```

Comment:
just inject the logger in the method if we need it, please :)
alternatively, you can pass it in the context

Reply:
Done. I did not inject it earlier because I was going to remove it after we resolve the issue.